ironfish-cli/src/commands/evm/shield.ts

```diff
@@ -0,0 +1,122 @@
+import { Flags, ux } from '@oclif/core'
+import { ethers } from 'ethers'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+
+export class ShieldCommand extends IronfishCommand {
+  static description = `Shield asset from EVM to private token`
+
+  static flags = {
+    ...LocalFlags,
+    amount: Flags.integer({
+      char: 'a',
+      description: 'The amount of the asset to shield',
+    }),
+    contractAddress: Flags.string({
+      char: 'c',
+      description: 'The EVM contract address of the asset to shield',
+    }),
+    nonce: Flags.integer({
+      char: 'n',
+      description: 'The nonce of the EVM transaction',
+    }),
+    to: Flags.string({
+      char: 't',
+      description: 'The Ironfish public address of the recipient',
+    }),
+    from: Flags.string({
+      char: 'f',
+      description: 'The account name of the sender',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(ShieldCommand)
+    let { amount, contractAddress, nonce, fee, to, privateKey, from } = flags
+    const client = await this.sdk.connectRpc()
+
+    if (!amount) {
+      const input = await ux.prompt('Enter amount to send: ', {
+        required: true,
+      })
+      amount = parseInt(input)
+    }
+
+    if (!nonce) {
+      const input = await ux.prompt('Enter nonce for EVM transaction: ', {
+        required: true,
+      })
+      nonce = parseInt(input)
+    }
+
+    if (!fee) {
+      const input = await ux.prompt('Enter fee for the transaction: ', {
+        required: true,
+      })
+      fee = parseInt(input)
+    }
+
+    if (!to) {
+      const input = await ux.prompt('Enter recipient address: ', {
+        required: true,
+      })
+      to = input
+    }
+
+    if (!from) {
+      const input = await ux.prompt('Enter account name: ', {
+        required: true,
+      })
+      from = input
+    }
+
+    if (!privateKey) {
+      const input = await ux.prompt('Enter private key of the sender account: ', {
+        required: true,
+      })
+      privateKey = EthUtils.prefix0x(input)
+    }
+
+    if (!contractAddress) {
+      const input = await ux.prompt(
+        'Enter ERC20 contract address you want to shield assets for: ',
+        {
+          required: true,
+        },
+      )
+      contractAddress = EthUtils.prefix0x(input)
+    }
+    const isIron = contractAddress === GLOBAL_CONTRACT_ADDRESS.toString()
+    const response = await client.wallet.getAccountPublicKey({ account: from })
+    const publicAddress = response.content.evmPublicAddress
+    if (!publicAddress) {
+      this.error(`Account ${from} does not have an EVM public address`)
+    }
+
+    let data: string
+    if (isIron) {
+      const globalContract = new ethers.Interface(ContractArtifact.abi)
+      data = globalContract.encodeFunctionData('shield_iron', [Buffer.from(to, 'hex')])
+    } else {
+      const contract = new ethers.Interface(GoldTokenJson.abi)
```

Comment:
we don't need to address this now, but I'm wondering if we could use a generic ERC20 abi here?

Reply:
I wasn't sure how to do this, that is why I created an example contract. As long as any ERC20 uses the same shield interface, the encoded data should be fine since the function signature is the same. But yea I agree long term that has to be something that is slightly cleaner.